### ISSUE TITLE
An exception occured: Access violation

### ISSUE THREAD

**Issue by Nightwing0815** [sieukrem/jn-npp-plugin#104]
Hi,

just installed the plugin and get following error:
![1](https://user-images.githubusercontent.com/54181775/107426175-2e9b7e80-6b20-11eb-866f-b156f6270da3.png)

`Notepad++ v7.9.1   (64-bit)
Build time : Nov  2 2020 - 01:07:46
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS Name : Windows 7 Ultimate (64-bit) 
OS Build : 7601.0
Current ANSI codepage : 1252
Plugins : 	CodeAlignmentNpp.dll 
			ComparePlugin.dll 
			DSpellCheck.dll jN.dll 
			JSMinNPP.dll 
			mimeTools.dll 
			NppConverter.dll 
			NppEditorConfig.dll 
			NppExport.dll 
`
What needs to be done?

THX in advance,

ToM

**Comment by sieukrem** [comment]
It could be an incompatibility with some of other plugins. Can you try to uninstall other plugins, one by one!

**Comment by Nightwing0815** [comment]
Of course. At least deinstalled all plugins, except this one, the error is still there.
I'll try it on my office PC later today...

ToM

**Comment by sieukrem** [comment]
Which version of jN?

**Comment by Nightwing0815** [comment]
`Notepad++ v7.9   (64-bit)
Build time : Sep 22 2020 - 03:19:04
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Pro for Workstations (64-bit) 
OS Version : 1909
OS Build : 18363.959
Current ANSI codepage : 1252
Plugins : 	CodeAlignmentNpp.dll	14.1.107
			ComparePlugin.dll		2.0.1
			DSpellCheck.dll			1.4.17
			NppEditorConfig.dll		0.4
			jN.dll				2.2.185.6
			NPPJSONViewer.dll 		1.40
			mimeTools.dll			2.5
			NppExport.dll			0.2.9
`

On my office computer, there's no error.
So I'll post you later the same from my home pc...

ToM

**Comment by Nightwing0815** [comment]
A bit late, but here it comes :-)

`Notepad++ v7.9.1   (64-bit)
Build time : Nov  2 2020 - 01:07:46
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS Name : Windows 7 Ultimate (64-bit) 
OS Build : 7601.0
Current ANSI codepage : 1252
Plugins :
CodeAlignmentNpp.dll			14.1.107
ComparePlugin.dll				2.0.1
DSpellCheck.dll					1.4.19
MarkdownViewerPlusPlus.dll		0.8.2
mimeTools.dll					2.5
nppConverter.dll				4.2.1
NppEditorConfig.dll				0.4
NppExport.dll					0.2.9
NPPJSONViewer.dll 				4.2.1
jN.dll							2.2.185.6`

Settings from my home pc, where the error occures...

ToM

**Comment by sieukrem** [comment]
The only one thing I see is the Npp version. And you still have other plugins in your environment

**Comment by sieukrem** [comment]
So I tried out v7.9.1 in my environment, no crashe. Please try to remove plugins.
Pluginmanger does not offer installation of 
codealignmentnpp
Nppjsonviewer
So I can not check them. For security reasons I do not want to install plugins which are listed in pm.

**Comment by Nightwing0815** [comment]
> And you still have other plugins in your environment

This is an interesting fact...
Let me look, reinstall, delete all data on weekend, just arrived home from work...

ToM

**Comment by Nightwing0815** [comment]
Okay, just ran a few setups: complete deinstall, reinstall --> nothing changed.
Loaded the portable version, same...
So i think, this might be a problem with my win7x64 installation @ home pc, cause on win10x64 on my work pc, everything is fine.
I'd say, let's wait to an update from npp, or do you have another idea?

ToM

**Comment by sieukrem** [comment]
You can check if there is crash dump. Usually npp creates one in npps root directory. If you find one then attach it to request

**Comment by sieukrem** [comment]
@Nightwing0815, do have any progress?

**Comment by Nightwing0815** [comment]
No, a crash dump wasn't there, so i had no chance for further investigation.
I just deinstalled it, cause i haven't the time, to dig deeper...

ToM

**Comment by Kpax7** [comment]
Hello there
I can confirm that I have the same problem with the latest versions of npp and jN:
---------------------------
Plugin Exception
---------------------------
An exception occurred due to plugin: jN.dll
Exception reason: Access violation
---------------------------
OK   
---------------------------

I also tried everything including uninstall, reinstall, update, removing everyother plugin,...
Here's the debugging info:

`Notepad++ v8.4.5   (32-bit)
Build time : Sep  3 2022 - 03:56:37
Path : C:\Program Files (x86)\Notepad++_8.4.5\notepad++.exe
Command Line : 
Admin mode : ON
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 7 Ultimate (64-bit) 
OS Build : 7601.0
Current ANSI codepage : 1256
Plugins : 
    jN (2.2.185.8)
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)
`

**Comment by yerpj** [comment]
Same here, with version 8.4.5 32-bit.
Installed plugins: 
- DSpellCheck 1.4.6
- jN Notepad++ Plugin 2.2.185.6
- MIME Tools 2.8
- Converter 4.4
- NppExport 0.4

**Comment by sieukrem** [comment]
It is again windows 7. 
1. Can you please look for a dump file?
2. Which action do you do so that it crashes?
3. Can you uninstall other plugins?

**Comment by Kpax7** [comment]
Thanks for the reply.
Windows 7 is still widely used and works well with other things

1. Yeah I looked at npp installation folder and no dump is created automatically, as @Nightwing0815  has already stated earlier.

> No, a crash dump wasn't there, so i had no chance for further investigation. I just deinstalled it, cause i haven't the time, to dig deeper...
> 
> ToM

2. There is no particular action. If you just install the plugin from the plugin admin or manually through copying it to the plugins folder, you get the error message upon the npp startup.

3. I totally uninstalled every other plugin. No luck. Uninstalled every plugin including jN and installed jN again. No luck.

Is there any debug/verbouse mode or error log in notepad++ to see what is actually happening?
Is there any dependency that has been added to the jN in the 2021 versions that may be lacking in Windows 7 installations?

**Comment by yerpj** [comment]
> It is again windows 7.

By the way, I am experiencing that error on **Windows 10** Enterprise (10.0.18363)

**Comment by onolox** [comment]
I have this problem, I think it's related to highlight, when I double click the file attached in random places the error appear. Specially the line 203 in "addFlowTowerTest".
dump attached.


[CalibrationShapes.zip](https://github.com/sieukrem/jn-npp-plugin/files/11051394/CalibrationShapes.zip)
[NppDump.zip](https://github.com/sieukrem/jn-npp-plugin/files/11051403/NppDump.zip)

**Comment by sieukrem** [comment]
@onolox I have built plugin with debug symbols, so that I can see more details in the dump file. Can you please try out following builds [x64](https://ci.appveyor.com/project/sieukrem/jn-npp-plugin/build/job/wbkyhlgvm2wrtjkq/artifacts) [x86](https://ci.appveyor.com/project/sieukrem/jn-npp-plugin/build/job/mkuna4iukgux65oi/artifacts) and then upload the dump file

**Comment by onolox** [comment]
[File004_utf8.zip](https://github.com/sieukrem/jn-npp-plugin/files/11054907/File004_utf8.zip)
[notepad14596.zip](https://github.com/sieukrem/jn-npp-plugin/files/11054915/notepad14596.zip)

**Comment by onolox** [comment]
I tried installing notepad++ v8.5 and intalling the plugins again, and bring the configs, no exception is happening now.

**Comment by onolox** [comment]
Alright! No more errors! n++ x86.
I saw that you have increased the size of the icons in the bar too!

**Comment by Nightwing0815** [comment]
Just a bit too late for me, I don't use windows any more :-)
Sadly, there's no npp for Linux...

ToM
